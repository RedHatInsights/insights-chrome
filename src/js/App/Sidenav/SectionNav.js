import React from 'react';
import PropTypes from 'prop-types';
import { NavGroup } from '@patternfly/react-core';
import ExpandableNav from './ExpandableNav';
import WrenchIcon from '@patternfly/react-icons/dist/js/icons/wrench-icon';
import SecurityIcon from '@patternfly/react-icons/dist/js/icons/security-icon';
import TrendUpIcon from '@patternfly/react-icons/dist/js/icons/trend-up-icon';
import './SectionNav.scss';

const anisbleHackIds = ['savings-planner', 'automation-calculator', 'organization-statistics', 'job-explorer', 'clusters', 'notifications'];

const sectionTitleMapper = (id) =>
  ({
    operations: (
      <div>
        <WrenchIcon />
        Operations Insights
      </div>
    ),
    security: (
      <div>
        <SecurityIcon />
        Security Insights
      </div>
    ),

    business: (
      <div>
        <TrendUpIcon />
        Business Insights
      </div>
    ),

    insights: <div>Insights</div>,
  }[id] || '');

const SectionNav = ({ items, section, onClick, ...props }) => {
  if (items?.length > 0) {
    return (
      <NavGroup className="ins-c-section-nav" id={section} title={sectionTitleMapper(section)}>
        {items.map((item, key) => (
          <ExpandableNav
            key={item.id || key}
            {...props}
            {...item}
            onClick={(event, subItem) => {
              // TODO: Fix me! It adds parent that have been removed in Navigation.js:209
              if (anisbleHackIds.includes(item.id)) {
                return onClick(event, item, { id: 'insights' });
              }
              return item.subItems ? onClick(event, subItem, item) : onClick(event, item);
            }}
          />
        ))}
      </NavGroup>
    );
  }
  return (
    <ExpandableNav
      title={props.title}
      id={props.id}
      onClick={(event, subItem) => (props.subItems ? onClick(event, subItem, props) : onClick(event, props))}
      {...props}
    />
  );
};

SectionNav.propTypes = {
  items: PropTypes.array,
  subItems: PropTypes.array,
  section: (props, propName, component) => {
    if (!props[propName] && Object.prototype.hasOwnProperty.call(props, 'items')) {
      return new Error(
        `Prop "${propName}" is required if object has "items" prop. Invalid prop ${propName} supplied to ${component}. Validation failed.`
      );
    }
    if (props[propName] && typeof props[propName] !== 'string') {
      return new Error(
        `Failed prop type. Invalid prop ${propName} supplied to ${component}. Expected "string", got "${typeof props[propName]}". Validation failed.`
      );
    }
  },
  id: PropTypes.string,
  title: PropTypes.string,
  activeLocation: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired,
};

export default SectionNav;
