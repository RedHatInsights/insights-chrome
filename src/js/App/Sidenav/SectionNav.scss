
.ins-c-app-title {
  color: var(--pf-global--Color--light-100);
  font-size: var(--pf-global--FontSize--lg);
  margin: 0 !important;
  white-space: nowrap;
}

.pf-c-nav__list {
  section.ins-c-section-nav {
    margin-top: 0 !important;
    h2.pf-c-nav__section-title {
      color: var(--pf-global--disabled-color--100);
      font-size: var(--pf-global--FontSize--xs);
      font-weight: var(--pf-global--FontWeight--semi-bold);
      text-transform: uppercase;
      padding-bottom: var(--pf-global--spacer--md);
      padding-left: var(--pf-global--spacer--lg);
      padding-top: var(--pf-global--spacer--md);
      }
    ul
      > li.pf-c-nav__item {
        margin-top: 0;
        border: var(--pf-global--BorderWidth--sm) var(--pf-global--BackgroundColor--dark-200) solid;
        border-left: 0;
        border-right: 0;
        border-top: 0;
        a.pf-c-nav__link, button.pf-c-nav__link {
          padding-left: var(--pf-global--spacer--xl);
          svg {
            margin-left: 3px;
          }
        }
        .pf-c-nav__subnav {
          --pf-c-nav__link--active--after--BorderColor: var(--pf-global--primary-color--light-100);
          --pf-c-nav__link--hover--after--BorderColor: var(--pf-global--primary-color--light-100);
          padding-left: var(--pf-global--spacer--xl);
          padding-bottom: 0;
          .pf-c-nav__list {
            .pf-c-nav__item {
              .pf-c-nav__link{
                padding-left: var(--pf-global--spacer--sm);
                svg {
                  width: 12px;
                  height: 12px;
                  margin-left: var(--pf-global--spacer--sm)
                }
              }
            }
          }
        }
        section.pf-c-nav__subnav {
          li.pf-c-nav__item {
            border: none !important;
          }
        }
      }
    }
    .ins-c-navigation__additional-links{
      --pf-c-nav__link--hover--after--BorderColor:  var(--pf-global--primary-color--light-100);
      a {
        padding-left: var(--pf-global--spacer--lg);
      }
      svg {
        width: 12px;
        height: 12px;
      }
    }
  }
