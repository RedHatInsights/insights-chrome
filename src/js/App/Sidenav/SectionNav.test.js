import React from 'react';
import SectionNav from './SectionNav';
import { render } from '@testing-library/react';

describe('SectionNav', () => {
  it('should render corectly', () => {
    const props = { id: 'id', title: 'title', activeLocation: 'loc', onClick: jest.fn() };
    const { container } = render(<SectionNav {...props} />);
    expect(container).toMatchSnapshot();
  });
  it('should render group with app', () => {
    const props = { id: 'id', title: 'title', activeLocation: 'loc', onClick: jest.fn(), items: [{ id: 'app', title: 'title' }] };
    const { container } = render(<SectionNav {...props} />);
    expect(container).toMatchSnapshot();
  });
  it('should render group with app and sub app', () => {
    const props = {
      id: 'id',
      title: 'title',
      activeLocation: 'loc',
      onClick: jest.fn(),
      items: [{ id: 'app', title: 'title', subItems: [{ id: 'subapp', title: 'title2' }] }],
    };
    const { container } = render(<SectionNav {...props} />);
    expect(container).toMatchSnapshot();
  });
  it('should render app without group', () => {
    const props = {
      id: 'app',
      title: 'title',
      activeLocation: 'loc',
      onClick: jest.fn(),
    };
    const { container } = render(<SectionNav {...props} />);
    expect(container).toMatchSnapshot();
  });
  it('should render app with subapp withou group', () => {
    const props = {
      id: 'app',
      title: 'title',
      activeLocation: 'loc',
      onClick: jest.fn(),
      subItems: [{ id: 'subapp', title: 'title2' }],
    };
    const { container } = render(<SectionNav {...props} />);
    expect(container).toMatchSnapshot();
  });
});
