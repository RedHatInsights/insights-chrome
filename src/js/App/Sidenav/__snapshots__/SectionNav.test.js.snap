// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SectionNav should render app with subapp and custom url, but without group  1`] = `
<div>
  <li
    class="pf-c-nav__item pf-m-expandable ins-m-navigation-align"
    data-ouia-component-id="app"
    data-ouia-component-type="PF4/NavExpandable"
    data-ouia-safe="true"
    itemid="app"
  >
    <button
      aria-expanded="false"
      class="pf-c-nav__link"
      id="app"
    >
      title
      <span
        class="pf-c-nav__toggle"
      >
        <span
          class="pf-c-nav__toggle-icon"
        >
          <svg
            aria-hidden="true"
            fill="currentColor"
            height="1em"
            role="img"
            style="vertical-align: -0.125em;"
            viewBox="0 0 256 512"
            width="1em"
          >
            <path
              d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
            />
          </svg>
        </span>
      </span>
    </button>
    <section
      aria-labelledby="app"
      class="pf-c-nav__subnav"
      hidden=""
    >
      
      <ul
        class="pf-c-nav__list"
      >
        <li
          class="pf-c-nav__item"
          data-ouia-component-id="subapp"
          data-ouia-component-type="PF4/NavItem"
          data-ouia-safe="true"
        >
          <a
            class="pf-c-nav__link"
            href="http://localhost/loc/app/subapp"
          >
            Title2
             
          </a>
        </li>
      </ul>
    </section>
  </li>
</div>
`;

exports[`SectionNav should render app with subapp without group 1`] = `
<div>
  <li
    class="pf-c-nav__item pf-m-expandable ins-m-navigation-align"
    data-ouia-component-id="app"
    data-ouia-component-type="PF4/NavExpandable"
    data-ouia-safe="true"
    itemid="app"
  >
    <button
      aria-expanded="false"
      class="pf-c-nav__link"
      id="app"
    >
      title
      <span
        class="pf-c-nav__toggle"
      >
        <span
          class="pf-c-nav__toggle-icon"
        >
          <svg
            aria-hidden="true"
            fill="currentColor"
            height="1em"
            role="img"
            style="vertical-align: -0.125em;"
            viewBox="0 0 256 512"
            width="1em"
          >
            <path
              d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
            />
          </svg>
        </span>
      </span>
    </button>
    <section
      aria-labelledby="app"
      class="pf-c-nav__subnav"
      hidden=""
    >
      
      <ul
        class="pf-c-nav__list"
      >
        <li
          class="pf-c-nav__item"
          data-ouia-component-id="subapp"
          data-ouia-component-type="PF4/NavItem"
          data-ouia-safe="true"
        >
          <a
            class="pf-c-nav__link"
            href="http://localhost/loc/app/subapp"
          >
            Title2
             
          </a>
        </li>
      </ul>
    </section>
  </li>
</div>
`;

exports[`SectionNav should render app without group 1`] = `
<div>
  <li
    class="pf-c-nav__item"
    data-ouia-component-id="app"
    data-ouia-component-type="PF4/NavItem"
    data-ouia-safe="true"
  >
    <a
      class="pf-c-nav__link"
      href="http://localhost/loc/app"
    >
      Title
       
    </a>
  </li>
</div>
`;

exports[`SectionNav should render corectly 1`] = `
<div>
  <li
    class="pf-c-nav__item"
    data-ouia-component-id="id"
    data-ouia-component-type="PF4/NavItem"
    data-ouia-safe="true"
  >
    <a
      class="pf-c-nav__link"
      href="http://localhost/loc/id"
    >
      Title
       
    </a>
  </li>
</div>
`;

exports[`SectionNav should render group with app 1`] = `
<div>
  <section
    aria-labelledby="section1"
    class="pf-c-nav__section ins-c-section-nav"
  >
    <h2
      class="pf-c-nav__section-title"
      id="section1"
    />
    <ul>
      <li
        class="pf-c-nav__item"
        data-ouia-component-id="app"
        data-ouia-component-type="PF4/NavItem"
        data-ouia-safe="true"
      >
        <a
          class="pf-c-nav__link"
          href="http://localhost/loc/app"
        >
          Title
           
        </a>
      </li>
    </ul>
  </section>
</div>
`;

exports[`SectionNav should render group with app and sub app 1`] = `
<div>
  <section
    aria-labelledby="section1"
    class="pf-c-nav__section ins-c-section-nav"
  >
    <h2
      class="pf-c-nav__section-title"
      id="section1"
    />
    <ul>
      <li
        class="pf-c-nav__item pf-m-expandable ins-m-navigation-align"
        data-ouia-component-id="app"
        data-ouia-component-type="PF4/NavExpandable"
        data-ouia-safe="true"
        itemid="app"
      >
        <button
          aria-expanded="false"
          class="pf-c-nav__link"
          id="app"
        >
          title
          <span
            class="pf-c-nav__toggle"
          >
            <span
              class="pf-c-nav__toggle-icon"
            >
              <svg
                aria-hidden="true"
                fill="currentColor"
                height="1em"
                role="img"
                style="vertical-align: -0.125em;"
                viewBox="0 0 256 512"
                width="1em"
              >
                <path
                  d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
                />
              </svg>
            </span>
          </span>
        </button>
        <section
          aria-labelledby="app"
          class="pf-c-nav__subnav"
          hidden=""
        >
          
          <ul
            class="pf-c-nav__list"
          >
            <li
              class="pf-c-nav__item"
              data-ouia-component-id="subapp"
              data-ouia-component-type="PF4/NavItem"
              data-ouia-safe="true"
            >
              <a
                class="pf-c-nav__link"
                href="http://localhost/loc/app/subapp"
              >
                Title2
                 
              </a>
            </li>
          </ul>
        </section>
      </li>
    </ul>
  </section>
</div>
`;

exports[`SectionNav should render group with app, sub app and custom url 1`] = `
<div>
  <section
    aria-labelledby="section1"
    class="pf-c-nav__section ins-c-section-nav"
  >
    <h2
      class="pf-c-nav__section-title"
      id="section1"
    />
    <ul>
      <li
        class="pf-c-nav__item pf-m-expandable ins-m-navigation-align"
        data-ouia-component-id="app"
        data-ouia-component-type="PF4/NavExpandable"
        data-ouia-safe="true"
        itemid="app"
      >
        <button
          aria-expanded="false"
          class="pf-c-nav__link"
          id="app"
        >
          title
          <span
            class="pf-c-nav__toggle"
          >
            <span
              class="pf-c-nav__toggle-icon"
            >
              <svg
                aria-hidden="true"
                fill="currentColor"
                height="1em"
                role="img"
                style="vertical-align: -0.125em;"
                viewBox="0 0 256 512"
                width="1em"
              >
                <path
                  d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
                />
              </svg>
            </span>
          </span>
        </button>
        <section
          aria-labelledby="app"
          class="pf-c-nav__subnav"
          hidden=""
        >
          
          <ul
            class="pf-c-nav__list"
          >
            <li
              class="pf-c-nav__item"
              data-ouia-component-id="subapp"
              data-ouia-component-type="PF4/NavItem"
              data-ouia-safe="true"
            >
              <a
                class="pf-c-nav__link"
                href="example.url.com"
              >
                Title2
                 
                <svg
                  aria-hidden="true"
                  fill="currentColor"
                  height="1em"
                  role="img"
                  style="vertical-align: -0.125em;"
                  viewBox="0 0 512 512"
                  width="1em"
                >
                  <path
                    d="M432,320H400a16,16,0,0,0-16,16V448H64V128H208a16,16,0,0,0,16-16V80a16,16,0,0,0-16-16H48A48,48,0,0,0,0,112V464a48,48,0,0,0,48,48H400a48,48,0,0,0,48-48V336A16,16,0,0,0,432,320ZM488,0h-128c-21.37,0-32.05,25.91-17,41l35.73,35.73L135,320.37a24,24,0,0,0,0,34L157.67,377a24,24,0,0,0,34,0L435.28,133.32,471,169c15,15,41,4.5,41-17V24A24,24,0,0,0,488,0Z"
                  />
                </svg>
              </a>
            </li>
          </ul>
        </section>
      </li>
    </ul>
  </section>
</div>
`;
